accessors/dataflow/dataflow_accessor.go

```diff
@@ -0,0 +1,94 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package dataflowacc
+
+import (
+	"context"
+	"fmt"
+
+	"cloud.google.com/go/dataflow/apiv1beta3/dataflowpb"
+	dataflowclient "github.com/GoogleCloudPlatform/spanner-migration-tool/accessors/clients/dataflow"
+	"github.com/GoogleCloudPlatform/spanner-migration-tool/logger"
+)
+
+type DataflowTuningConfig struct {
+	ProjectId             string            `json:"projectId"`
+	JobName               string            `json:"jobName"`
+	Location              string            `json:"location"`
+	VpcHostProjectId      string            `json:"hostProjectId"`
+	Network               string            `json:"network"`
+	Subnetwork            string            `json:"subnetwork"`
+	MaxWorkers            int32             `json:"maxWorkers"`
+	NumWorkers            int32             `json:"numWorkers"`
+	ServiceAccountEmail   string            `json:"serviceAccountEmail"`
+	MachineType           string            `json:"machineType"`
+	AdditionalUserLabels  map[string]string `json:"additionalUserLabels"`
+	KmsKeyName            string            `json:"kmsKeyName"`
+	GcsTemplatePath       string            `json:"gcsTemplatePath"`
+	AdditionalExperiments []string          `json:"additionalExperiments"`
+	EnableStreamingEngine bool              `json:"enableStreamingEngine"`
+}
+
+func GetDataflowLaunchRequest(parameters map[string]string, cfg DataflowTuningConfig) (*dataflowpb.LaunchFlexTemplateRequest, error) {
```

Comment:
What is the use case for a public method that just generates a request? The consumer of the client would be interested in either launching,updating,fetching attributes of or stopping the dataflow job. What would the consumer do with a request object?

Reply:
+1

Reply:
Makes sense, will move the launchRequest function to the caller, only functions calling a client should reside here.